### DISCUSSION TITLE
Greppo - python framework for building geospatial dashboards and web-apps

### DISCUSSION THREAD

**Discussion by krish-adi** [encode/starlette#1342] [Ideas]
Hey there Starlette users 👋,

Greppo repo - [https://github.com/greppo-io/greppo](https://github.com/greppo-io/greppo)

Greppo is a python framework for Data Scientists looking to prototype interactive Geospatial web-applications with Python. The library is a toolkit to quickly integrate data, algorithms, APIs and visualizations without any frontend code. It provides APIs to recompute the backend logic (a data mutation hook for the backend) and the updates are reflected in the frontend. All the worries of building backend, frontend and APIs taken care.

Here’s an example of an app that we built in a few minutes, and deployed using docker container service: [**app**](https://greppo-demo-sfo.sm48tr5o7beei.us-west-2.cs.amazonlightsail.com/). We visualize the public amenities (toilets, atm etc) in San Francisco, and display the count within a selected area. You can interact with the app by modifying the pre-selected area loaded by default on the app or adding a new one. On update, the bar chart should reflect the new counts across the selected areas.

To get started simply install with pip, `pip install greppo`.

We are actively developing Greppo, and would love to hear your feedback!

It uses Starlette ! So a shout out and thanks to the encode/starlette team 🙌 !

**Comment by Kludex** [comment]
PR welcome to include on our [Third party packages](https://www.starlette.io/third-party-packages/) list.

**Reply by krish-adi** [reply]
Submitted a PR. https://github.com/encode/starlette/pull/1343
